### PR TITLE
VIVI-8270 add px as a termux package

### PR DESCRIPTION
**Pull request by mattd233**

### CODE REVIEW

File: px/main.py [hunk 1]
```diff
@@ -598,6 +598,7 @@ def set_username(username):
     State.username = username
 
 def set_password():
+    sys.exit()
```
Comment (tristanpenman) [inline px/main.py]: To stick with the styling of commenting out code we don't want, can you comment out everything here up to the original `sys.exit()`?

File: px/main.py [hunk 2]
```diff
@@ -874,6 +875,8 @@ def parse_config():
 # Exit related
 
 def quit(checkOnly = False):
+    sys.exit()
+    return
```
Comment (tristanpenman) [inline px/main.py]: Same here

File: px/main.py [hunk 3]
```diff
@@ -1043,6 +1046,7 @@ def restore_stdout():
             State.debug.stdout = State.stdout
 
     def attach_console():
```
Comment (tristanpenman) [inline px/main.py]: In this case, we should probably comment out the origianl call to `attach_console()`

File: px/wproxy.py [hunk 4]
```diff
@@ -243,17 +243,18 @@ def find_proxy_for_url(self, url):
             return copy.deepcopy(self.servers), netloc, path
 
         if self.mode == MODE_CONFIG_PAC:
```
Comment (tristanpenman) [inline px/wproxy.py]: Could actually comment this too

File: px/main.py [hunk 5]
```diff
@@ -598,24 +598,25 @@ def set_username(username):
     State.username = username
 
 def set_password():
-    try:
-        if len(State.username) == 0:
-            pprint("domain\\username missing - specify via --username or configure in px.ini")
-            sys.exit()
-        pprint("Setting password for '" + State.username + "'")
+    sys.exit()
```
Comment (tristanpenman) [inline px/main.py]: Can you remove this line, and uncomment the original `sys.exit()`?

File: px/main.py [hunk 6]
```diff
@@ -874,69 +875,71 @@ def parse_config():
...
-            sys.stdout.flush()
-            time.sleep(4)
-            quit(checkOnly = True)
-    else:
-        if checkOnly:
-            pprint(" DONE")
-        else:
-            pprint("Px is not running")
-
     sys.exit()
```
Comment (tristanpenman) [inline px/main.py]: Same with this one

File: px/wproxy.py [hunk 7]
```diff
@@ -242,18 +242,18 @@ def find_proxy_for_url(self, url):
...
+            # if self.pac is None:
+            #     # Load PAC file
+            #     self.pac = Pac(debug_print = dprint)
+            #     if self.servers[0].startswith("http"):
+            #         self.pac.load_url(self.servers[0])
+            #     else:
+            #         self.pac.load_jsfile(self.servers[0])
+
+            # if self.pac is not None:
+            #     return parse_proxy(self.pac.find_proxy_for_url(url, netloc[0])), netloc, path
```
Comment (tristanpenman) [inline px/wproxy.py]: Only minor, but I think these comments should start one indentation level to the left

File: px/main.py [hunk 8]
```diff
@@ -874,67 +874,68 @@ def parse_config():
...
-            sys.stdout.write("Quitting Px ..")
-            sys.stdout.flush()
-            time.sleep(4)
-            quit(checkOnly = True)
-    else:
-        if checkOnly:
-            pprint(" DONE")
-        else:
-            pprint("Px is not running")
+    # return
```
Comment (tristanpenman) [inline px/main.py]: Leftover line

File: px/main.py [hunk 9]
```diff
@@ -472,10 +472,10 @@ def print_banner():
...
     )
 
-    if sys.platform == "win32":
-        if is_compiled() or "pythonw.exe" in sys.executable:
-            if State.config.getint("settings", "foreground") == 0:
-                detach_console()
+    # if sys.platform == "win32":
+    #     if is_compiled() or "pythonw.exe" in sys.executable:
+    #         if State.config.getint("settings", "foreground") == 0:
+                # detach_console()
```
Comment (tristanpenman) [inline px/main.py]: Could fix the comment alignment here, if we're being pedantic

**Review by tristanpenman — APPROVED** [review]
Some minor suggestions, but looks good otherwise. Happy to merge and work from there

**Comment by mattd233** [comment]
Waiting for junmin's branch to get approved first before I merge since this is based on that branch.

**Review by tristanpenman — APPROVED** [review]

**Pull request merged**